swift/Samples/Tutorial/Tutorial3.md

```diff
@@ -0,0 +1,645 @@
+
+```swift
+final class TodoEditViewController: ScreenViewController<TodoEditScreen> {
+    // ..rest of the implementation...
+
+    override func screenDidChange(from previousScreen: TodoEditScreen) {
+        update(with: screen)
+    }
+
+    private func update(with screen: TodoEditScreen) {
+        // Update the view with the data from the screen.
+        todoEditView.title = screen.title
+        todoEditView.note = screen.note
+        todoEditView.onTitleChanged = screen.onTitleChanged
+        todoEditView.onNoteChanged = screen.onNoteChanged
+    }
+
+}
+```
+
+Finally, register the `TodoEditScreen` in the `TutorialContainerViewController` with the viewRegistry:
+
+```swift
+public final class TutorialContainerViewController: UIViewController {
+    let containerViewController: UIViewController
+
+    public init() {
+        // Create a view registry. This will allow the infrastructure to map `Screen` types to their respective view controller type.
+        var viewRegistry = ViewRegistry()
+        // Register the `WelcomeScreen` and view controller with the convenience method the template provided.
+        viewRegistry.registerWelcomeScreen()
+        // Register the `TodoListScreen` and view controller with the convenience method the template provided.
+        viewRegistry.registerTodoListScreen()
+        // Register the `BackStackContainer`, which provides a container for the `BackStackScreen`.
+        viewRegistry.registerBackStackContainer()
+        // Register the `TodoEditScreen` and view controller with the convenience method the template provided.
+        viewRegistry.registerTodoEditScreen()
+
+        // Create a `ContainerViewController` with the `RootWorkflow` as the root workflow, with the view registry we just created.
+        containerViewController = ContainerViewController(
+            workflow: RootWorkflow(),
+            viewRegistry: viewRegistry)
+
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    // ... rest of the implementation ...
+```
+
+#### TodoEditWorkflow
+
+Now that we have our screen and view controller, update the `TodoEditWorkflow` to emit this screen as the rendering.
+
+The `TodoEditWorkflow` needs an initial Todo item passed into it from its parent. It will make a copy of it in its internal state - this can be the "scratch pad" for edits. This allows changes to be made and still be able to discard the changes if the user does not want to save them.
+
+Additionally, we will (finally) use the `workflowDidChange` method. If the edit workflow's parent provides an updated `todo`, it will invalidate the `todo` in `State`, and replace it with the one provided from the parent.
+
+```swift
+// MARK: Input and Output
+
+struct TodoEditWorkflow: Workflow {
+
+    // The "Todo" passed from our parent.
+    var initialTodo: TodoModel
+
+    enum Output {
+
+    }
+}
+
+
+// MARK: State and Initialization
+
+extension TodoEditWorkflow {
+
+    struct State {
```

Comment:
Do you even need this?

Reply:
Yes, we need a `State` definition here. It's holding the *copy* of the item being edited, which allows for it to be saved or discarded.